### ISSUE TITLE
Comparison breaking changes in 2.56.0

### ISSUE THREAD

**Issue by TMMeijers** [briannesbitt/Carbon#2544]
Hello,

I encountered an issue with the following code:
```php
Carbon\Carbon::now()->lessThan(null);
```

Breaking changes were introduced with #2530 as `null` is now resolved into a Carbon instance `lessThan(null)` returns a different result.

Carbon version: **2.56.0**

PHP version: **8.0.13**

I expected to get:

```
false
```

But I actually get:

```
true
```

Thanks!

**Comment by kylekatarnls** [comment]
Hello 👋 

I understand it can be handy to support nullable values here, but it was never really intended to.

I would recommend to avoid comparing `null` to any kind of thing.

There is no consensus of what should be the expected result:

```php
var_dump(new DateTime() < null);
> false

var_dump(new DateTime() > null);
> true
```

```sql
SELECT NOW() < NULL
> NULL

SELECT NOW() < NULL
> NULL
```

And the same applies to the next person reading your code. This person will have no particular reason to think it will return `false`.

You better can give the answer explicitly to the next reader or the future you with no ambiguity:
```php
$myVariable = null; // or a Carbon instance

if ($myVariable !== null && Carbon::now()->lessThan($myVariable)) {
    // This is a date in the future
}
```

Or even better:
```php
$myVariable = null; // or a Carbon instance

if ($myVariable instanceof CarbonInterface && $myVariable->isFuture()) {
    // This is a date in the future
}
```
And you get a code really clear about what it's doing without having to take any assertion or doing retro-engineering.

I will weight if it's worth to change it as it would be breaking for people that started to use the new version and may have asserted that `null` resolve to `now`. I would even be tempted to say the correct behavior should be to raise an exception when passing to such method a value that is not comparable to a date.

**Comment by kylekatarnls** [comment]
Yes, when those BC can be noticed shortly and show a high number of users impacted, I typically revert it as per the policy described here:
https://github.com/briannesbitt/Carbon/blob/master/contributing.md#note-about-semantic-versioning-and-breaking-changes

If a revert comes after few days and just fixes an edge-case, it just does more arm than good, it's actually an other BC for the people who started to use the new feature. And honestly I think the previous behavior was actually wrong and inconsistent with the rest of the methods, sometimes you would just not increase the major version, when your lib says 2 + 2 = 5, you just consider it a bug and hope no one took this result for granted.

Depending on the usage, every change is a breaking change, that's why on the top of semver (which has the convention not to introduce BC unless on major release, but it's actually not a universal rule in OSS, it's a proposition libraries may or may not adhere BTW), it's important to consider how BC is defined and how it can be spotted in a realistic way because there is no tag undo buttons for libraries . With hundreds millions downloads, there is no way to cover all the possible usages of the library (you can't afford 10k lines of test for each line of actual code) and anticipate all the potential impacts. We provide instead unit tests covering 100% of the code showing the supported uses and also check the examples given in the doc which aims to provide a wide-enough surface for behavior change detection.

**Comment by AlbinoDrought** [comment]
Is it possible to retroactively update the release notes at https://github.com/briannesbitt/Carbon/releases/tag/2.56.0 about this change?

**Comment by kylekatarnls** [comment]
Added.
